src/Context/MongoDbContext.cs

```diff
@@ -61,11 +59,19 @@ public IMongoCollection<TDocument> CreateCollection<TDocument>()
 
         private void EnsureInitialized()
         {
-            Initialize(MongoOptions);
+            if (_mongoDbContextData == null)
```

Comment:
The double lock pattern is not really necessary. Except, you want to exclude the case, that if the initializing is exactly in this moment ongoing, then its okay.

Reply:
It's better for the client to don't throw in a race condition. Once initialized it will not hit the lock anymore